update_deps/propagate_updates.ps1

```diff
@@ -116,151 +107,38 @@ function update-local-repo {
-        $check_runs = $check_run_content.check_runs
-        if($check_runs.Length -eq 0) {
-            # retry if check-runs have not started yet
-            continue
-        }
-        for($i=0; $i -lt $check_runs.Length; $i++) {
-            $check = $check_runs[$i]
-            # if check has not completed, wait and retry
-            if($check.status -ne "completed") {
-                break
-            }
-            # if check has complete but was not successful, throw error
-            elseif ($check.conclusion -ne "success") {
-                Write-Error "Checks are failing for PR: $html_url"
-                exit -1
-            }
-            # if all checks have completed and were successful, exit while loop
-            if($i -eq ($check_runs.Length - 1)) {
-                # timeout to let mergeable state stabilize
-                spin 10
-                $get_pr_response = Invoke-WebRequest -URI $api_url -UseBasicParsing -Method Get -Headers $github_header
-                if(!$get_pr_response) {
-                    Write-Error "Failed to get PR: $html_url"
-                    exit -1
-                }
-                $get_pr_response = $get_pr_response.Content | ConvertFrom-Json
-                # PR should be in mergeable_state "clean"
-                # if it is not, something unexpected happened and error should be thrown
-                if($get_pr_response.mergeable_state -ne "clean"){
-                    Write-Host "Unexpected mergeable state `"$($get_pr_response.mergeable_state)`" for PR: $html_url"
-                    exit -1
-                }
-                return
-            }
-        }
-    }
-}
-
-
-# merge Github PR into master
-function merge-pr-github {
-    param(
-        [string] $api_url,
-        [string] $html_url
-    )
-    $request_url = $api_url + "/merge"
-    $body = @{
-        merge_method="squash"
-    }
-    $body = $body | ConvertTo-Json
-    $response = Invoke-WebRequest -URI $request_url -UseBasicParsing -Method Put -Headers $github_header -Body $body
-    if(!$response) {
-        Write-Error "Failed to merge PR: $html_url"
-        exit -1
-    }
-    $content = $response.Content | ConvertFrom-Json
-    # if PR fails to merge, throw error
-    if(!$content.merged){
-        Write-Error "Failed to merge PR: $html_url"
-        Write-Error $content
-        exit -1
-    }
-}
-
-
 # update dependencies for Github repo
 function update-repo-github {
     param(
-        [string] $repo_name
+        [string] $repo_name,
+        [string] $new_branch_name
     )
+    cd $repo_name
     Write-Host "`nCreating PR"
-    $api_url, $html_url = create-pr-github $repo_name
+    $working_directory = (Get-Location).Path
+    gh pr create --title "[autogenerated] update dependencies" --body "Propagating dependency updates" --head $new_branch_name
+    Write-Host "Waiting for checks to start"
+    Start-Sleep -Seconds 30 
```

Comment:
>Start-Sleep -Seconds 30  [](http://example.com/codeflow?start=4&length=24)

curious why a 30s wait is needed...but probably doesn't matter because none of our gates likely can run any faster than that...

Reply:
It takes a moment for the checks to start. If `gh pr checks --watch` is called before the checks are started, it can skip over the checks that have not been started.

The 30s sleep allows enough time for all the checks to start